src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -568,7 +594,32 @@ int MPL_gavl_tree_delete(MPL_gavl_tree_t gavl_tree, const void *addr, uintptr_t
     };
 
   fn_exit:
-    return mpl_err;
-  fn_fail:
-    goto fn_exit;
+    return;
+}
+
+static void gavl_tree_purge_nodes(gavl_tree_s * tree_ptr, uintptr_t addr, uintptr_t len)
+{
```

Comment:
The `gavl_tree_purge_nodes` function name is not straightforward. Can we rename it with a more explicit name such as `delete_removed_nodes`?

Reply:
It is better to add a function description saying it deletes intersect nodes.